app/models/ddr/batch/batch_object_role.rb

```diff
@@ -0,0 +1,36 @@
+module Ddr::Batch
+
+  class BatchObjectRole < ActiveRecord::Base
+    belongs_to :batch_object, :inverse_of => :batch_object_roles
+
+    OPERATION_ADD = "ADD".freeze  # Add the principal and role to the object in the indicated scope
+    OPERATIONS = [ OPERATION_ADD ].freeze
+
+    validates :operation, inclusion: { in: OPERATIONS }
+    with_options if: :operation_requires_agent? do |obj|
+      obj.validates :agent, presence: true
+    end
+    with_options if: :operation_requires_role_type? do |obj|
+      obj.validates :role_type, inclusion: { in: Ddr::Auth::Roles.role_types.map(&:title) }
+    end
+    with_options if: :operation_requires_role_scope? do |obj|
+      obj.validates :role_scope, inclusion: { in: Ddr::Auth::Roles::SCOPES }
```

Comment:
According to http://reservedwords.herokuapp.com/, "scope" is a reserved word in Ruby on Rails.  Whether using it here would create an actual conflict, I can't say for sure.  I'd prefer to err on the cautious side and stick with :role_scope.